### ISSUE TITLE
High Memory Consumption of json.Marshal Leading to OOM

### ISSUE THREAD

**Issue by pc-quiknode** [Fantom-foundation/Sonic#190]
**Version**
```
$ sonicd version
Sonic
Version: 1.2.1-e
Git Commit: a6f79ac99065b347b678f66446697737d2fe87f6
Git Commit Date: 1721218173
Architecture: amd64
Protocol Versions: [63]
Go Version: go1.21.6
Operating System: linux
```

**Describe the bug**
We have observed an issue that memory usage of Sonic nodes could spike intermittently even with low GOMEMLIMIT and GOGC:
![Screenshot from 2024-08-09 12-27-05](https://github.com/user-attachments/assets/860ec948-a308-4039-9f1e-7d7dfe2021ec)

Here's a pprof heap dump showing that json.Marshal could consume dozens of gigabytes making RPC responses:
![ftm-mainnet-archive-lsn-fra-12155 heap](https://github.com/user-attachments/assets/8934c437-45f6-43c1-ac50-d73c64328d24)

**Comment by thaarok** [comment]
Should be resolved by #193

**Comment by pc-quiknode** [comment]
We no longer see spikes in memory usage.
Thanks
